tests/integrationv2/providers.py

```diff
@@ -225,21 +218,26 @@ def _join_ciphers(self, ciphers):
     def _cipher_to_cmdline(self, protocol, cipher):
         cmdline = list()
 
-        if cipher.min_version is Protocols.TLS13:
-            cmdline.append('-ciphersuites')
-        else:
-            cmdline.append('-cipher')
-
         ciphers = []
         if type(cipher) is list:
+            # In the case of a cipher list we need to be sure TLS13 specific ciphers aren't
+            # mixed with ciphers from previous versions
+            version = cipher[0].min_version
+            is_tls13 = (version == Protocols.TLS13)
+            mismatch = [c for c in cipher if (c.min_version is Protocols.TLS13) != is_tls13]
+
+            if len(mismatch) > 0:
+                raise Exception("Cannot combine TLS1.3 ciphers with older ciphers: {}".format([c.name for c in cipher]))
+
             ciphers.append(self._join_ciphers(cipher))
         else:
-            if cipher.min_version == Protocols.TLS13:
-                ciphers.append(OpenSSL.supported_ciphers[cipher])
-            else:
-                # This replace is only done for ciphers that are not TLS13 specific.
-                # Run `openssl ciphers` to view the inconsistency in naming.
-                ciphers.append(cipher.name.replace("_", "-"))
+            version = cipher.min_version
+            ciphers.append(cipher.name)
+
+        if version is Protocols.TLS13:
```

Comment:
It would be cleaner if you set `is_tls13` instead of `version` inside the if/else, and just checked `is_tls13` here. Also, we generally try to do >= instead of == to not cause problems for theoretical future us and our theoretical future 1.4 project.